src/natcap/invest/coastal_vulnerability.py

```diff
@@ -2599,27 +2724,24 @@ def _aggregate_raster_values_in_radius(
            [False,  True,  True,  True,  True,  True, False],
            [False, False, False,  True, False, False, False]])
 
+    Do the aggregation with an arbitrary function ``aggregation_op``.
+
     Args:
         base_point_vector_path (string): point vector with projected
             coordinates in units matching sample_disatnce.
         base_raster_path (string): raster file with square pixels and
             projection matching base_point_vector_path.
         sample_distance (float): radius around each point to search
             for valid pixels.
-        aggregation_mode (string): 'mean' or 'density'. Calculate the mean
-            of valid pixels in search window, or if 'density', divide the mean
-            by the area of valid pixels. 'density' assumes units of meters
-            for sample_distance and raster geotransform.
         target_pickle_path (string): path to pickle file storing dict
             keyed by point id.
+        aggregation_op (function): takes a signal array and a mask array
+            of same shape. Returns scalar.

```

Comment:
Or returns `numpy.nan` if no masked signal pixels are found?

Reply:
Not strictly. The habitat op actually returns 5 in that case, representing no habitat found. Because we're allowing nodata to represent known absence of habitat. As I'm writing that out, it starts to sound like a bad idea. Hmm.

Reply:
Sorry I was splitting hairs here. I added "or numpy.nan". But I think we can leave it up to the actual `op` to describe when it might return `numpy.nan`, as it might vary by the `op`.